lib/AST/ASTVerifier.cpp

```diff
@@ -2114,11 +2114,20 @@ class Verifier : public ASTWalker {
     }
 
     void verifyChecked(ValueDecl *VD) {
-      if (!VD->hasAccess() && !VD->getDeclContext()->isLocalContext() &&
-          !isa<GenericTypeParamDecl>(VD) && !isa<ParamDecl>(VD)) {
-        dumpRef(VD);
-        Out << " does not have access";
-        abort();
+      if (VD->hasAccess()) {
+        if (VD->getFormalAccess() == AccessLevel::Open &&
```

Comment:
Could add a similar check for 'final'

Reply:
I'm scared that'll turn up more places where it's wrong, but I guess that's the point.